packages/admin-frontend/components/edit_book/EditBookForm.js

```diff
@@ -0,0 +1,177 @@
+import FormHelperText from '@material-ui/core/FormHelperText';
+
+import { saveBook } from '../../lib/fetch';
+import type { BookDetails } from '../../types';
+
+const PUBLISHING_STATUS = ['PUBLISHED', 'FLAGGED', 'UNLISTED'];
+const PAGE_ORIENTATIONS = ['PORTRAIT', 'LANDSCAPE'];
+
+type Props = {
+  book: BookDetails
+};
+
+type State = {
+  book: BookDetails
+};
+
+export default class EditBookForm extends React.Component<Props, State> {
+  state = {
+    book: this.props.book
+  };
+
+  handleSubmit = (content: BookDetails) => {
+    this.updateBook(content);
+  };
+
+  updateBook = async (content: BookDetails) => {
+    const result = await saveBook(content);
+    if (result.isOk) {
+      this.setState({ book: content });
+    }
+  };
+
+  render() {
+    const book = this.state.book;
+
+    return (
+      <div>
+        {' '}
+        {book && (
+          <Typography variant="headline" component="h1" gutterBottom>
+            Editing book:{' '}
+            <Link href={`/${book.language.code}/books/details/${book.id}`}>
+              <a>{book.title}</a>
+            </Link>
+          </Typography>
+        )}
+        <Form
+          initialValues={book}
+          onSubmit={this.handleSubmit}
+          validate={handleValidate}
+          render={({ handleSubmit, pristine, form, invalid }) => (
+            <form>
+              <Grid container spacing={24} direction="column">
+                <Grid item xs>
+                  <Field
+                    name="title"
+                    render={({ input, meta }) => (
+                      <>
+                        <TextField
+                          fullWidth
+                          required
+                          label="Title"
+                          {...input}
+                        />
+                        {meta.error &&
+                          meta.touched && (
+                            <FormHelperText error>{meta.error}</FormHelperText>
+                          )}
+                      </>
+                    )}
+                  />
+                </Grid>
+
+                <Grid item xs>
+                  <Field
+                    name="description"
+                    render={({ input, meta }) => (
+                      <>
+                        <TextField
+                          fullWidth
+                          required
+                          label="Description"
+                          {...input}
+                          multiline
+                        />
+                        {meta.error &&
+                          meta.touched && (
+                            <FormHelperText error>{meta.error}</FormHelperText>
+                          )}
+                      </>
+                    )}
+                  />
+                </Grid>
+
+                <Grid item xs>
+                  <Field
+                    name="pageOrientation"
+                    render={({ input }) => (
+                      <>
```

Comment:
The fragment wrapping the select here is redudant. We should also label the field.

Reply:
fixed